cardinal/plugin_persona.go

```diff
@@ -221,18 +234,18 @@ func buildPersonaIndex(wCtx engine.Context) (personaIndex, error) {
 				return true
 			}
 			lowerPersona := strings.ToLower(sc.PersonaTag)
-			personaTagToAddress[lowerPersona] = personaIndexEntry{
+			globalPersonaTagToAddressIndex[lowerPersona] = personaIndexEntry{
```

Comment:
Is index the right word for this? This looks like memoization. 

A general memoization module owned by eCtx or world would be more palatable for reviewers. 

In this case it can live independent the concept of an index as it's a much more simpler concept. 

I personally don't care. It can live as a global, but in this case it must be marked very loudly as temporary. If not then a general memoization module surrounding certain logic and owned by world/eCtx works too.

Reply:
The name matches the terminology for the already-existing struts.

A more general memoization module in wCtx would be more involved and requires a design pass. This PR is meant to fix a specific performance problem.

Reply:
Highly disagree. This PR itself fails a design pass like a red flag, so even a application specific memoization module or even moving these variables out of the global scope is better than what you did here. Idc to much though if this will be removed. I'll leave it to Scott to do the final approval.